src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -321,26 +321,36 @@ private bool CanInstallPackage()
         {
             if (DisplayRoots != null)
             {
-                foreach (PackageItem packageItem in DisplayRoots)
+                CheckIfFilesAreSelected(DisplayRoots);
+            }
+
+            return !_isInstalling && !NoFilesSelected;
+        }
+
+        private void CheckIfFilesAreSelected(IReadOnlyList<PackageItem> children)
```

Comment:
>CheckIfFilesAreSelected [](start = 21, length = 23)

By name, I wouldn't have thought this method would have changed state, but instead would return a bool.

Reply:
Renamed this method and refactored the code. I need to maintain state for NoFilesSelected as bind this in InstallDialogXaml to refresh state for the tree view for files

---
In reply to: [194555082](https://github.com/aspnet/LibraryManager/pull/126#discussion_r194555082) [](ancestors = 194555082)

Reply:
Why not have the caller change the state if that's what's desired.

---
In reply to: [194565600](https://github.com/aspnet/LibraryManager/pull/126#discussion_r194565600) [](ancestors = 194565600,194555082)

Reply:
I renamed the field, the negations were confusing. Sorry

---
In reply to: [194567317](https://github.com/aspnet/LibraryManager/pull/126#discussion_r194567317) [](ancestors = 194567317,194565600,194555082)